src/library/dlgtagfetcher.cpp

```diff
@@ -270,6 +329,52 @@ void DlgTagFetcher::apply() {
                 trackRelease.releaseGroupId);
     }
 #endif // __EXTRA_METADATA__
+
+    if (!m_fetchedCoverArtByteArrays.isNull()) {
+        VERIFY_OR_DEBUG_ASSERT(m_isCoverArtCopyWorkerRunning == false) {
+            return;
+        };
+
+        QFileInfo trackFileInfo = QFileInfo(m_track->getLocation());
+
+        QString coverArtCopyFilePath =
+                trackFileInfo.absoluteFilePath().left(
+                        trackFileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
+                "jpeg";
```

Comment:
Thank you for the feedback @ronso0 

> why jpeg? can we be sure about that?

When I first implement this i used "jpg" because all of the responses were "jpg" on the links, that worked fine. After implementation of cover art copy worker, there was an error because the actual file extension was "jpeg" and on the name it was "jpg".  So I changed to jpeg.

> Idk if that's faster but it's more readable IMO.

That looks way better obviously. But with this change when we download the cover art there is one slash "/" is missing and that will download the cover art in wrong path. 

In my case the file paths are:
Old file path = "/home/username/Desktop/music/supertrouper.jpeg"
New file path =  "/home/username/Desktop/musicsupertrouper.jpeg"

When I implement it at first time, I wanted to use something easy to read in terms of file path, such as (canonicalpath() + '/' + trackBaseName + ".jpeg") but then I realized that might not work for windows since the file paths are separated with '\', I'm not sure about other operating systems.

So because of the various file path formats, I just deleted the extension of the track and added jpeg. I couldn't find a better way to compose the cover art file path, is there a better way to do this?

Reply:
Oh sorry, I didn't notice the missing dir separator.

Anyway, adding the suffix manually might not even be required. Please correct me if I'm wrong:
That you saw only `jpg` until now and no official info (I found) that musicbrainz only stores jpg... means we could get other formats and run into a format mismatch.
And then [`ImageFileData::saveFile`](https://github.com/mixxxdj/mixxx/blob/9965d55f6e00cd1982f78cf09ac6196887869487/src/util/imagefiledata.cpp#L52) would throw an error but use the potentially wrong suffix anyway?
That's probably
> When I first implement this i used "jpg" because all of the responses were "jpg" on the links, that worked fine. After implementation of cover art copy worker, there was an error because the actual file extension was "jpeg" and on the name it was "jpg".

So, why specify the format at all when `ImageFileData` can figure out the correct extension from the fetched bytes array https://github.com/mixxxdj/mixxx/blob/9965d55f6e00cd1982f78cf09ac6196887869487/src/util/imagefiledata.cpp#L12-L16 
Did I overlook something?

Btw [m_oldCoverArtFilePath](https://github.com/mixxxdj/mixxx/blob/9965d55f6e00cd1982f78cf09ac6196887869487/src/library/export/coverartcopyworker.h#L27) should actually be m_**new**CoverArtFilePath, since that's the file that is written, no?

Reply:
Thanks for the review! There is no overlook, everything is fine. So no worries! 🐙 

I think everything is solved under this comment, except the [m_oldCoverArtFilePath](https://github.com/mixxxdj/mixxx/blob/9965d55f6e00cd1982f78cf09ac6196887869487/src/library/export/coverartcopyworker.h#L27) . 

It is actually both new and old cover art file path. Because we have an existing cover art (old) and user is choosing new cover art (selected), then selected is replaced at where the old cover art is. So as I said it can be considered as both new and old. I'm not sure though. Should we rename it under this PR (Since I changed the related file on this PR already that will be ok IMHO), or this can be done in separate PR (A Pr with not big changes would make the PR history more crowded), or just not rename it at all. What do you think @ronso0 @daschuer ?